src/domains/services/validation.service.ts

```diff
@@ -32,6 +32,154 @@ import { ParseOptions } from '@asyncapi/parser';
 import { ParserOptions } from '@asyncapi/parser/cjs/parser';
 import { calculateScore } from '@/utils/scoreCalculator';
 
+import { ConfigService } from './config.service';
+
+// GitHub API response type
+interface GitHubFileInfo {
+  download_url: string;
+  content?: string;
+  encoding?: string;
+  name: string;
+  path: string;
+  sha: string;
+  size: number;
+  type: string;
+}
+
+// GitHub host constants
+const GITHUB_HOSTS = [
+  'raw.githubusercontent.com', // eslint-disable-line sonarjs/no-duplicate-string
+  'github.com', // eslint-disable-line sonarjs/no-duplicate-string
+  'api.github.com' // eslint-disable-line sonarjs/no-duplicate-string
+];
+
+/**
+ * Helper function to validate if a URL is a GitHub blob URL
+ */
+const isValidGitHubBlobUrl = (url: string): boolean => {
+  try {
+    const parsedUrl = new URL(url);
+    return parsedUrl.hostname === 'github.com' && parsedUrl.pathname.includes('/blob/');
+  } catch (error) {
+    return false;
+  }
+};
+
+/**
+ * Helper function to validate if a URL is a GitHub URL (any GitHub host)
+ */
+const isValidGitHubUrl = (url: string): boolean => {
+  try {
+    const parsedUrl = new URL(url);
+    return GITHUB_HOSTS.includes(parsedUrl.hostname);
+  } catch (error) {
+    return false;
+  }
+};
+
+/**
+ * Custom GitHub URL resolver for private repositories
+ */
+const createHttpWithAuthResolver = () => ({
+  schema: 'https',
+  order: 1,
+
+  canRead: (uri: any) => {
+    try {
+      const url = new URL(uri.toString());
+      
+      // Check if host is github.com
+      if (url.hostname !== 'github.com') {
+        return false;
+      }
+      
+      // Check if path contains /blob/
+      return url.pathname.includes('/blob/');
+    } catch (error) {
+      return false;
+    }
+  },
+
+  /**
+   * Convert GitHub web URL to API URL
+   */
+  convertGitHubWebUrl: (url: string): string => {
+    // Remove fragment from URL before processing
+    const urlWithoutFragment = url.split('#')[0];
+    
+    // Handle GitHub web URLs like: https://github.com/owner/repo/blob/branch/path
+    // eslint-disable-next-line no-useless-escape
+    const githubWebPattern = /^https:\/\/github\.com\/([^\/]+)\/([^\/]+)\/blob\/([^\/]+)\/(.+)$/;
+    const match = urlWithoutFragment.match(githubWebPattern);
+    
+    if (match) {
+      const [, owner, repo, branch, path] = match;
+      return `https://api.github.com/repos/${owner}/${repo}/contents/${path}?ref=${branch}`;
+    }
+    return url;
+  },
+
+  read: async (uri: any) => {
+    let url = uri.toString();
+    
+    // Default headers
+    const headers: Record<string, string> = {
+      'User-Agent': 'AsyncAPI-CLI'
+    };
+
+    const authInfo = await ConfigService.getAuthForUrl(url);
+
+    if (isValidGitHubBlobUrl(url)) {
+      url = createHttpWithAuthResolver().convertGitHubWebUrl(url);
+    }
+    
+    // Only require authentication for GitHub URLs
```

Comment:
This comment is not true. We should not require authentication **only** for GitHub URLs.

Reply:
Make sense thanks